### ISSUE TITLE
Problem with Fine-tuning GLIP - classes

### ISSUE THREAD

**Issue by j-sheikh** [open-mmlab/mmdetection#11014]
Hello!

thank you for your nice work!

I am currently trying to fine tune GLIP for my custom dataset. I have converted my data to CocoDataset type and set up a config file. 
I am able to train the model, but when evaluating, one class (class_4) is not recognized when I enter the class as a text prompt. It works for all the others. I have checked my dataset/json files and everything looks good. My categories look like this [{'id': cat_id, 'name': label, 'supercategory': label}. In the config file I also added:
 classes = ('class_1', ''class_2', ''class_3', ''class_4', ''class_5', ''class_6', 'class_7'), num_classes = len(classes). 

I use the following setup:
_base_ = '../glip_atss_swin-t_b_fpn_dyhead_16xb2_ms-2x_funtune_coco.py'
load_from = '../glip_tiny_c_mmdet-2fc427dd.pth'

The problem is that class_4 is not recognized, the result is empty:

<DetDataSample(

    META INFORMATION
    ori_shape: (200, 200)
    img_id: 0
    img_path: 'xxx'
    custom_entities: False
    pad_shape: (800, 800)
    scale_factor: (4.0, 4.0)
    batch_input_shape: (800, 800)
    img_shape: (800, 800)
    text: 'class_4'

    DATA FIELDS
    pred_instances: <InstanceData(
        
            META INFORMATION
        
            DATA FIELDS
            labels: tensor([], device='cuda:0', dtype=torch.int64)
            bboxes: tensor([], device='cuda:0', size=(0, 4))
            scores: tensor([], device='cuda:0')
        ) at 0x7f11d09afca0>
    ignored_instances: <InstanceData(
        
            META INFORMATION
        
            DATA FIELDS
            labels: tensor([], device='cuda:0', dtype=torch.int64)
            bboxes: tensor([], device='cuda:0', size=(0, 4))
        ) at 0x7f11d09ad840>
    token_positive_map: 
    gt_instances: <InstanceData(
        
            META INFORMATION
        
            DATA FIELDS
            labels: tensor([], device='cuda:0', dtype=torch.int64)
            bboxes: tensor([], device='cuda:0', size=(0, 4))
        ) at 0x7f11d09afac0>
) at 0x7f11d09aded0>

I am thankful for any help or recommendation. 
Cheers

**Comment by hhaAndroid** [comment]
@j-sheikh Thank you for your trial. For your situation, you may first want to visualize the JSON to see if it can be correctly parsed by the COCO API. Then, you can run the `tools/analysis_tools/browse_dataset.py` script to visualize and check if the dataset is in proper order. If everything appears normal, please provide the complete configuration for further investigation into the issue.

**Comment by j-sheikh** [comment]
@hhaAndroid Hey, thanks for the response. I investigated my dataset format and run the browse_dataset.py. The results suggest that my dataset is in proper form. Class 4 is also visualized. Here is my full config:

config_content = """
_base_ = '/content/mmdetection/configs/glip/glip_atss_swin-t_b_fpn_dyhead_16xb2_ms-2x_funtune_coco.py'
load_from = '/content/drive/MyDrive/GLIP/glip_tiny_c_mmdet-2fc427dd.pth'
lang_model_name = 'bert-base-uncased'

data_root = '/content/drive/MyDrive/GLIP/dataset/'
classes = ('class_1', 'class_2', 'class_3', 'class_4', 'class_5', 'class_6', 'class_7')
num_classes = len(classes)
work_dir = '/content/drive/MyDrive/GLIP/checkpoints_and_logs_4'

train_batch_size_per_gpu = 2
#train_num_workers = 2
#persistent_workers = True #False if num_worker is 0

val_batch_size_per_gpu = 2
#val_num_workers = 2

max_epochs = 3



model = dict(
    type='GLIP',
    data_preprocessor=dict(
        type='DetDataPreprocessor',
        mean=[103.53, 116.28, 123.675],
        std=[57.375, 57.12, 58.395],
        bgr_to_rgb=False,
        pad_size_divisor=32),
    backbone=dict(
        type='SwinTransformer',
        embed_dims=96,
        depths=[2, 2, 6, 2],
        num_heads=[3, 6, 12, 24],
        window_size=7,
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        drop_rate=0.,
        attn_drop_rate=0.,
        drop_path_rate=0.2,
        patch_norm=True,
        out_indices=(1, 2, 3),
        with_cp=False,
        convert_weights=False),
    neck=dict(
        type='FPN_DropBlock',
        in_channels=[192, 384, 768],
        out_channels=256,
        start_level=0,
        relu_before_extra_convs=True,
        add_extra_convs='on_output',
        num_outs=5),
    bbox_head=dict(
        type='ATSSVLFusionHead',
        lang_model_name=lang_model_name,
        num_classes=num_classes,
        in_channels=256,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            ratios=[1.0],
            octave_base_scale=8,
            scales_per_octave=1,
            strides=[8, 16, 32, 64, 128],
            center_offset=0.5),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoderForGLIP',
            target_means=[.0, .0, .0, .0],
            target_stds=[0.1, 0.1, 0.2, 0.2]),
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='GIoULoss', loss_weight=2.0),
        loss_centerness=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)),
    language_model=dict(type='BertModel', name=lang_model_name),
    train_cfg=dict(
        assigner=dict(
            type='ATSSAssigner',
            topk=9,
            iou_calculator=dict(type='BboxOverlaps2D_GLIP')),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.6),
        max_per_img=100))

# dataset settings
train_pipeline = [
    dict(
        type='LoadImageFromFile',
        imdecode_backend='pillow',
    ),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='GTBoxSubOne_GLIP'),
    dict(
        type='RandomChoiceResize',
        scales=[(1333, 480), (1333, 560), (1333, 640), (1333, 720),
                (1333, 800)],
        keep_ratio=True,
        resize_type='FixScaleResize',
        backend='pillow'),
    dict(type='RandomFlip_GLIP', prob=0.5),
    dict(type='FilterAnnotations', min_gt_bbox_wh=(1, 1)),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor', 'flip', 'flip_direction', 'text',
                   'custom_entities'))
]

test_pipeline = [
    dict(
        type='LoadImageFromFile',
        imdecode_backend='pillow'),
    dict(
        type='FixScaleResize',
        scale=(800, 1333),
        keep_ratio=True,
        backend='pillow'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor', 'text', 'custom_entities'))
]

train_dataloader = dict(
    batch_size=train_batch_size_per_gpu,
    dataset=dict(
        _delete_=True,
        type='RepeatDataset',
        times=2,
        dataset=dict(
            type='CocoDataset',
            metainfo=dict(classes=classes),
            data_root = data_root,
            ann_file='train/annotations/train_annotations.json',
            data_prefix=dict(img='train/'),

            pipeline=train_pipeline,
            return_classes=True,
        )))

val_dataloader = dict(
    batch_size=val_batch_size_per_gpu,
    dataset=dict(
        type='CocoDataset',
        metainfo=dict(classes=classes),
        data_root=data_root,
        ann_file='val/annotations/val_annotations.json',
        data_prefix=dict(img='val/'),
        pipeline=test_pipeline,
        return_classes=True,
    ))

test_dataloader = dict(
    batch_size=val_batch_size_per_gpu,
    dataset=dict(
        type='CocoDataset',
        data_root=data_root,
        metainfo=dict(classes=classes),
        ann_file='test/annotations/test_annotations.json',
        data_prefix=dict(img='test/'),
        pipeline=test_pipeline,
        return_classes=True,
    ))

# Reduce evaluation time
val_evaluator = dict(
    type='mmdet.CocoMetric',
    proposal_nums=(100, 1, 10),
    ann_file=data_root + 'val/annotations/val_annotations.json',
    metric='bbox')

test_evaluator = dict(
    type='mmdet.CocoMetric',
    proposal_nums=(100, 1, 10),
    ann_file= data_root + 'test/annotations/test_annotations.json',
    metric='bbox')

# We did not adopt the official 24e optimizer strategy
# because the results indicate that the current strategy is superior.
optim_wrapper = dict(
    _delete_=True,
    type='OptimWrapper',
    optimizer=dict(
        type='AdamW', lr=0.00002, betas=(0.9, 0.999), weight_decay=0.05),
    paramwise_cfg=dict(
        custom_keys={
            'absolute_pos_embed': dict(decay_mult=0.),
            'relative_position_bias_table': dict(decay_mult=0.),
            'norm': dict(decay_mult=0.)
        }),
    clip_grad=None)

"""

with open("/content/config.py", "w") as file:
    file.write(config_content)

**Comment by j-sheikh** [comment]
@hhaAndroid i followed your suggestions and it seems everything is fine. When running browse_dataset.py class 4 is visible in the images (bbox). Here is my full config

```python
config_content = """
_base_ = '/content/mmdetection/configs/glip/glip_atss_swin-t_b_fpn_dyhead_16xb2_ms-2x_funtune_coco.py'
load_from = '/content/drive/MyDrive/GLIP/glip_tiny_c_mmdet-2fc427dd.pth'
lang_model_name = 'bert-base-uncased'

data_root = '/content/drive/MyDrive/GLIP/dataset/'
classes = ('class_1', 'class_2', 'class_3', 'class_4', 'class_5', 'class_6', 'class_7')
num_classes = len(classes)
work_dir = '/content/drive/MyDrive/GLIP/checkpoints_and_logs_4'

train_batch_size_per_gpu = 2


val_batch_size_per_gpu = 2


max_epochs = 3



model = dict(
    type='GLIP',
    data_preprocessor=dict(
        type='DetDataPreprocessor',
        mean=[103.53, 116.28, 123.675],
        std=[57.375, 57.12, 58.395],
        bgr_to_rgb=False,
        pad_size_divisor=32),
    backbone=dict(
        type='SwinTransformer',
        embed_dims=96,
        depths=[2, 2, 6, 2],
        num_heads=[3, 6, 12, 24],
        window_size=7,
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        drop_rate=0.,
        attn_drop_rate=0.,
        drop_path_rate=0.2,
        patch_norm=True,
        out_indices=(1, 2, 3),
        with_cp=False,
        convert_weights=False),
    neck=dict(
        type='FPN_DropBlock',
        in_channels=[192, 384, 768],
        out_channels=256,
        start_level=0,
        relu_before_extra_convs=True,
        add_extra_convs='on_output',
        num_outs=5),
    bbox_head=dict(
        type='ATSSVLFusionHead',
        lang_model_name=lang_model_name,
        num_classes=num_classes,
        in_channels=256,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            ratios=[1.0],
            octave_base_scale=8,
            scales_per_octave=1,
            strides=[8, 16, 32, 64, 128],
            center_offset=0.5),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoderForGLIP',
            target_means=[.0, .0, .0, .0],
            target_stds=[0.1, 0.1, 0.2, 0.2]),
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='GIoULoss', loss_weight=2.0),
        loss_centerness=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)),
    language_model=dict(type='BertModel', name=lang_model_name),
    train_cfg=dict(
        assigner=dict(
            type='ATSSAssigner',
            topk=9,
            iou_calculator=dict(type='BboxOverlaps2D_GLIP')),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.6),
        max_per_img=100))


train_pipeline = [
    dict(
        type='LoadImageFromFile',
        imdecode_backend='pillow',
    ),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='GTBoxSubOne_GLIP'),
    dict(
        type='RandomChoiceResize',
        scales=[(1333, 480), (1333, 560), (1333, 640), (1333, 720),
                (1333, 800)],
        keep_ratio=True,
        resize_type='FixScaleResize',
        backend='pillow'),
    dict(type='RandomFlip_GLIP', prob=0.5),
    dict(type='FilterAnnotations', min_gt_bbox_wh=(1, 1)),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor', 'flip', 'flip_direction', 'text',
                   'custom_entities'))
]

test_pipeline = [
    dict(
        type='LoadImageFromFile',
        imdecode_backend='pillow'),
    dict(
        type='FixScaleResize',
        scale=(800, 1333),
        keep_ratio=True,
        backend='pillow'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor', 'text', 'custom_entities'))
]

train_dataloader = dict(
    batch_size=train_batch_size_per_gpu,
    dataset=dict(
        _delete_=True,
        type='RepeatDataset',
        times=2,
        dataset=dict(
            type='CocoDataset',
            metainfo=dict(classes=classes),
            data_root = data_root,
            ann_file='train/annotations/train_annotations.json',
            data_prefix=dict(img='train/'),

            pipeline=train_pipeline,
            return_classes=True,
        )))

val_dataloader = dict(
    batch_size=val_batch_size_per_gpu,
    dataset=dict(
        type='CocoDataset',
        metainfo=dict(classes=classes),
        data_root=data_root,
        ann_file='val/annotations/val_annotations.json',
        data_prefix=dict(img='val/'),
        pipeline=test_pipeline,
        return_classes=True,
    ))

test_dataloader = dict(
    batch_size=val_batch_size_per_gpu,
    dataset=dict(
        type='CocoDataset',
        data_root=data_root,
        metainfo=dict(classes=classes),
        ann_file='test/annotations/test_annotations.json',
        data_prefix=dict(img='test/'),
        pipeline=test_pipeline,
        return_classes=True,
    ))

val_evaluator = dict(
    type='mmdet.CocoMetric',
    proposal_nums=(100, 1, 10),
    ann_file=data_root + 'val/annotations/val_annotations.json',
    metric='bbox')

test_evaluator = dict(
    type='mmdet.CocoMetric',
    proposal_nums=(100, 1, 10),
    ann_file= data_root + 'test/annotations/test_annotations.json',
    metric='bbox')


optim_wrapper = dict(
    _delete_=True,
    type='OptimWrapper',
    optimizer=dict(
        type='AdamW', lr=0.00002, betas=(0.9, 0.999), weight_decay=0.05),
    paramwise_cfg=dict(
        custom_keys={
            'absolute_pos_embed': dict(decay_mult=0.),
            'relative_position_bias_table': dict(decay_mult=0.),
            'norm': dict(decay_mult=0.)
        }),
    clip_grad=None)

"""

with open("/content/config.py", "w") as file:
    file.write(config_content)


```

**Comment by merlinxu-zz** [comment]
I met the same problem!! I defined 56 categories, but the model could only recognize 53 categories,;  the remaining 3 categories were AP=0 during inference.
